docs/server_config/nginx.conf.dist

```diff
@@ -0,0 +1,180 @@
+# -------------------------------------------------------------------------------------------------
+# The following file is provided as-is by PrestaShop as an example configuration for your Nginx server.
+# It may be incomplete, and remember you must adapt it for your own server's needs!
+# Based on the work of MattLoyeD and Michael Dekker.
+# -------------------------------------------------------------------------------------------------
+
+server {
+    # Ipv4
+    listen 80;
+
+    # IPv6
+    # listen [::]:80;
+
+    # SSL Ipv4 & v6
+    # listen 443 ssl;
+    # listen [::]:443 ssl;
+
+    # Your SSL Certificates, don't forget to take a look at Certbot (https://certbot.eff.org)
+    # ssl_certificate /etc/ssl/fullchain.pem;
+    # ssl_certificate_key /etc/ssl/privkey.pem;
+
+    # ssl_session_timeout 24h;
+    # ssl_session_cache shared:SSL:10m;
+    # ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
+    # ssl_ciphers ECDH+AESGCM:ECDH+AES256:ECDH+AES128:DH+3DES:RSA+3DES:AES128-SHA:!ADH:!AECDH:!MD5;
+    # ssl_prefer_server_ciphers on;
+    # Do not forget to create this file before with OpenSSL : "openssl dhparam -out /etc/nginx/ssl/dhparam.pem 2048"
+    # ssl_dhparam /etc/nginx/ssl/dhparam.pem;
+
+    # [REQUIRED EDIT] Your domain name goes here
+    server_name server_name example.com www.example.com;
+
+    # [REQUIRED EDIT] Absolute path to your website root on the filesystem
+    root /path/to/prestashop;
+
+    index index.php
+
+    # Redirect 404 errors to prestashop
+    error_page 404 /index.php?controller=404;
+
+    # Gzip Settings, convert all types.
+    gzip on;
+    gzip_vary on;
+    gzip_proxied any;
+
+    # Can be enhance to 5, but it can slow you server
+    # gzip_comp_level    5;
+    # gzip_min_length    256;
+
+    gzip_types
+        application/atom+xml
+        application/javascript
+        application/json
+        application/ld+json
+        application/manifest+json
+        application/rss+xml
+        application/vnd.geo+json
+        application/vnd.ms-fontobject
+        application/x-font-ttf
+        application/x-web-app-manifest+json
+        application/xhtml+xml
+        application/xml
+        font/opentype
+        image/bmp
+        image/svg+xml
+        image/x-icon
+        text/cache-manifest
+        text/css
+        text/plain
+        text/vcard
+        text/vnd.rim.location.xloc
+        text/vtt
+        text/x-component
+        text/x-cross-domain-policy;
+
+    gzip_disable "MSIE [1-6]\.(?!.*SV1)";
+
+    # Static assets delivery optimisations
+    add_header Strict-Transport-Security max-age=31536000;
```

Comment:
Thanks @inetbiz ! I'll fix that.

On the other hand, I now realize that the comment was wrong, I'll comment this out by default to avoid problems with people who aren't using HTTPS.